### ISSUE TITLE
Telnet password on 1.5.6_0001

### ISSUE THREAD

**Issue by osigida** [AlexxIT/XiaomiGateway3#1223]
This is not an issue, but the way to share info: 

I've got a new Gateway 3 with `1.5.5_000?` firmware installed. Unfortunately, I got it updated to 1.5.6_0001 before I realized there were issues with setting up. 
HA was not able to connect with the error `wrong_telnet`. 
I've enabled telnet by 5-2-2-2-2-2 clicks on the device as described on https://www.youtube.com/shorts/qJ21wWNE1Gg
Then telnet became available, but it required a password.  
And `admin/admin` worked 🎉 
then I just followed https://github.com/zvldz/mgl03_fw/tree/main/firmware#the-easy-way and installed 1.5.4 version. 
And then successfully added the hub into HA 

Hope, it is useful, feel free to close the ticket. 

Thanks a lot for the plugin!!

**Comment by uavana** [comment]
great job

**Comment by zhaoyingpu** [comment]
This method is effective for me. After successfully telnet gateway 3 with 1.5.6_0001,  'cat /data/miio/device.conf' to obtain the key.

**Comment by AlexxIT** [comment]
As I know, this won't work with all gateways from the market.

**Comment by apacukaa** [comment]
Hi, I have the same Gateway3, with 1.5.6_0001, never used for in HA, but now i would like to. I enabled telnet, tried to connect to, but i get "login incorrect" message. Im i doing something wrong or mine has other password? I tried "123123" and without password too. What else should i try?

**Comment by osigida** [comment]
@apacukaa you may try to brute-force the password, something tells me it should be something simple.

**Comment by dandymsk** [comment]
Good day! i need help with the same problem.
i have a telnet access to GW 3 - but credientals - admin/admin, root/root and etc... doesn't work.
i want to add GW to HomeAssistant (i have token and telnet error...)
![image](https://github.com/AlexxIT/XiaomiGateway3/assets/96170484/34eae67e-4e07-4d96-8d93-2ee2973b93eb)

**Comment by osigida** [comment]
@dandymsk, if you do have telnet access, try to install 1.5.4 FW as explained here: https://github.com/zvldz/mgl03_fw/tree/main/firmware#the-easy-way

**Comment by ss109** [comment]
牛x的密码。成功解决了降级的问题。

**Comment by dandymsk** [comment]
> 牛x的密码。成功解决了降级的问题。
@ss109 
Please help me. what **login** do you use ?
i've try - 
login - **admin**
password - **Bull's eye / Bullseye / bullseye** 
but it still doesn't work ...(((

**Comment by osigida** [comment]
@dandymsk I saw two options, login `admin` and either no password at all or `admin` as the password

**Comment by p1ratrulezzz** [comment]
admin/admin didn't work for me unfortunately... Got login incorrect error

**Comment by osigida** [comment]
@p1ratrulezzz dare to run some brute-force script? 
I have 3 gateways, one had no password (some old fw), other 2 were on 1.5.5.006 with the password `admin`

**Comment by p1ratrulezzz** [comment]
yep) I'll run the script and will use UART if no luck. for now i consider
buying a new hub in case if i brick this one 😄

**Comment by AlexxIT** [comment]
Brute-force will last until the end of the universe. I've been checking

**Comment by p1ratrulezzz** [comment]
yes, even if password is 6 random chars, it will last forever. So looks
like uart is the jedi path

**Comment by osigida** [comment]
Maybe a dictionary then? I doubt it is random.
I also found `app` as user name and no password on some forum as an option

**Comment by p1ratrulezzz** [comment]
> Maybe a dictionary then? I doubt it is random.
> I also found `app` as user name and no password on some forum as an option

app didn't work either. As for dictionary attack I'll try to use weakpass password list maybe, good idea even if will take a month )

**Comment by FilipeOS** [comment]
I think I just broke my gateway 3... I did that 5-2-2-2-2-2 in my 1.4.xxx and it's blinking orange and don't go out... Any advise? I tried since telnet was not working admin/app or whatever....

**Comment by alexpts** [comment]
I have mi gateway 3 with v 1.56_0001.  Method 5-2-2-2-2-2 help me to open telnet port (23). 

I was able to log admin/admin via telnet

**Comment by p1ratrulezzz** [comment]
No other methods helped at all... I had to solder UART and found key in the boot logs so I was able to add it to home assistant using key + token (from mihome vevs app). 

> I think I just broke my gateway 3... I did that 5-2-2-2-2-2 in my 1.4.xxx and it's blinking orange and don't go out... Any advise? I tried since telnet was not working admin/app or whatever....

I believe UART is also variant to unbrick the device, but in your case it looks like you just need to try to press 10 times the button again and it will reset the device, then just reconnect it in mi home. Orange light normally means "in pairing mode" waiting for adding to mi home, probably you didn't wait enough in 5-2-2-2-2 method beause instead of each dash there should be a pause for 1-2 sec

**Comment by FilipeOS** [comment]
But in video there's no 1/2 seconds wait.. I think I'll wait for some other option for noobs to update it....

**Comment by p1ratrulezzz** [comment]
no, definetely you should wait after each iterations

5 quick clicks, pause. 2 quick clicks, pause, 2 quick clicks, pause ... and
so on until the led blinks once meaning that telnet enabled

**Comment by FilipeOS** [comment]
I can't make it work on 1.4.6_0043 - EU.
I do the 5-2-2-2-2-2 and nothing happens. When I access via putty to ip, ask me the login and I tried admin, app, xiaomi, mi, etc with blank pw and can't never access.... any alternative please?

**Comment by AlexxIT** [comment]
This clicks works only on 1.5.5+ firmwares. But lower firmwares don't need clicks. Integration can remove telnet password on first connect.

**Comment by FilipeOS** [comment]
But whn I try to add gateway always say wrong telnet command, and I don't know password? Please advise

**Comment by AlexxIT** [comment]
You can try to download old HassWP versions with this old firmware support.

**Comment by FilipeOS** [comment]
I don't understand, why should I install a portable home assistant if I want only to update my gateway 3 to work with latest firmware available here? Sorry

**Comment by AlexxIT** [comment]
It's easiest way to remove telnet password. But you can search other instructions

**Comment by p1ratrulezzz** [comment]
> I don't understand, why should I install a portable home assistant if I want only to update my gateway 3 to work with latest firmware available here? Sorry

Alex is right, on your old firmware telnet password will be removed after adding to home assistant. The easiest way ever. You can install temporarly home assistant using docker, add your gateway and get your password removed.

Another solution is to use miio.php client -https://github.com/skysilver-lab/php-miio -havent tested it but should work. Check example of usage here https://github.com/AlexxIT/XiaomiGateway3/wiki/Decode-Telnet-Password

**Comment by FilipeOS** [comment]
So, I need to run https://github.com/AlexxIT/HassWP in my Win11 (for example), install, add gateway and remove/uninstall again and add again in my actual HA? Sorry to ask

**Comment by FilipeOS** [comment]
I've installed HASSWP and tried to add the xiaomi gateway 3 integration but I get same error when trying to configure via cloud or token:

Name: Gateway
Model: lumi.gateway.mgl03
IP: 192.168.XX.XXX
MAC: XX:EF:XX:XX:XX:BF
Token: XXX6774eXXXXXX547a4b424XXXXXX54a
Telnet: wrong_telnet

Please advise... Europe server

**Comment by AlexxIT** [comment]
You can try more old HassWP versions

**Comment by Yura0427** [comment]
купив з прошивкою 1.5.6_0001 , telnet був заблокований, натискання кнопки живлення 5-2-2-2-2-2 відкрило telnet(але не з першого разу - зробив спроб 10-20, постійно скидало до стандартних налаштувань), тепер не можу підіпрати пароль , пробував admin/-, admin/admin, app/-, app/app, root/-, root/root,  нічого не підходить, дайте знати якщо хтось зайде новий пароль, або спосіб як його дізнатися

**Comment by p1ratrulezzz** [comment]
> купив з прошивкою 1.5.6_0001 , telnet був заблокований, натискання кнопки живлення 5-2-2-2-2-2 відкрило telnet(але не з першого разу - зробив спроб 10-20, постійно скидало до стандартних налаштувань), тепер не можу підіпрати пароль , пробував admin/-, admin/admin, app/-, app/app, root/-, root/root, нічого не підходить, дайте знати якщо хтось зайде новий пароль, або спосіб як його дізнатися

вот то же самое было, в итоге помолого только воткнуть UART и считать ключ из логов. возможно, что с новыми прошивками там у всех рандомные пароли. Ни один стандартный не подошел. Тут совет только реально UART припаять, сходить к корешу с паяльником попросить припаять 4 провода. Взять с алика стандартный ch341a программатор и с него подключиться. В итоге этот метод оказался самым простым. Ну я в итоге просто шлейф припаял и через корпус его вытащил наружу, то есть девайс сейчас закрыт, но из него торчат проводки для программатора на всякий. можно конечно открыть прочитать и снова закрыть. Другие способы имхо бесполезны тут, к сожалению

**Comment by YoRHaGreen** [comment]
As I understand it, there are no new solutions with firmware 1.5.6_0001, only via UART?
